app/components/primer/icon_button.rb

```diff
@@ -55,11 +84,5 @@ def initialize(icon:, scheme: DEFAULT_SCHEME, box: false, **system_arguments)
 
       validate_aria_label
```

Comment:
It looks like `Primer::IconButton` requires an `aria-label` , but if tooltip slot is set with `type: :label`, an `aria-labelledby` association is created so we no longer need an `aria-label`. We should update this validation logic. Then we may want to update the comments in `@accessibility`.